src/nodes/ListItem.js

```diff
@@ -128,12 +128,14 @@ export default class ListItem extends TiptapListItem {
 						return false
 					}
 
-					tr.setNodeMarkup(parentList.pos, schema.nodes.list_item, { type: parentList.node.attrs.type === TYPES.CHECKBOX ? TYPES.BULLET : TYPES.CHECKBOX })
+					state.doc.nodesBetween(selection.from, selection.to, (node, pos) => {
+						if (node.type === schema.nodes.list_item) {
+							tr.setNodeMarkup(pos, node.type, { type: parentList.node.attrs.type === TYPES.CHECKBOX ? TYPES.BULLET : TYPES.CHECKBOX })
+						}
+					})
 					tr.scrollIntoView()
 
-					if (dispatch) {
-						dispatch(tr)
-					}
+					if (dispatch) dispatch(tr)
```

Comment:
Not sure about our coding style convention here, I thought that we prefer to always use curly brackets for better readability. But I might be wrong :)

Reply:
Yes, I'd prefer that, though not enforced through our eslint rules currently. 

https://github.com/nextcloud/eslint-config/blob/master/index.js
https://eslint.org/docs/rules/curly